tests/test_pages.py

```diff
@@ -33,6 +34,166 @@ def to_item(self) -> dict:
+
+    assert item1.text == "a"
+    assert item2.text == "b"
+
+
+@pytest.mark.asyncio
+async def test_multi_layout_page_object_shared_partial_layout():
+    """Scenario where a multi-layout page object acts as a switch for 2 or
+    more layout page objects that all inherit from some other page object class
+    that implements extraction for shared fields."""
+
+    @attrs.define
+    class PartialItem:
+        url: str
+
+    @attrs.define
+    class FullItem(PartialItem):
+        text: str
+
+    class PartialPage(WebPage[PartialItem]):
+        @field
+        async def url(self) -> str:
+            return str(self.response.url)
+
+    class FullPage1(PartialPage, Returns[FullItem]):
+        @field
+        async def text(self) -> Optional[str]:
+            return self.css("h1::text").get()
+
+    class FullPage2(PartialPage, Returns[FullItem]):
+        @field
+        async def text(self) -> Optional[str]:
+            return self.css("h2::text").get()
+
+    @attrs.define
+    class MyMultiLayoutPage(MultiLayoutPage[FullItem]):
+        response: HttpResponse
+        page1: FullPage1
+        page2: FullPage2
+
+        async def get_layout(self) -> ItemPage[FullItem]:
+            if self.response.css("h1::text"):
+                return self.page1  # type: ignore[return-value]
+            return self.page2  # type: ignore[return-value]
+
+    html1 = b"""
+    <!DOCTYPE html>
+    <html lang="en">
+        <head>
+            <title>h1</title>
+        </head>
+        <body>
+            <h1>a</h1>
+        </body>
+    </html>
+    """
+    html2 = b"""
+    <!DOCTYPE html>
+    <html lang="en">
+        <head>
+            <title>h2</title>
+        </head>
+        <body>
+            <h2>b</h2>
+        </body>
+    </html>
+    """
+
+    url = "https://example.com"
+    response1 = HttpResponse(url, body=html1)
+    page1_1 = FullPage1(response=response1)
+    page2_1 = FullPage2(response=response1)
+    response2 = HttpResponse(url, body=html2)
+    page1_2 = FullPage1(response=response2)
+    page2_2 = FullPage2(response=response2)
+
+    multilayoutpage1 = MyMultiLayoutPage(
+        response=response1, page1=page1_1, page2=page2_1
+    )
+    multilayoutpage2 = MyMultiLayoutPage(
+        response=response2, page1=page1_2, page2=page2_2
+    )
+
+    # To access page object fields, you must first get the underlying page
+    # object, and then access its fields:
+    layout1 = await multilayoutpage1.get_layout()
+    assert await layout1.url == url
+    assert await layout1.text == "a"
+    layout2 = await multilayoutpage2.get_layout()
+    assert await layout2.url == url
+    assert await layout2.text == "b"
+
+    # Returned items work as expected.
+    item1 = await multilayoutpage1.to_item()
+    assert item1.url == url
+    assert item1.text == "a"
+    item2 = await multilayoutpage2.to_item()
+    assert item2.url == url
+    assert item2.text == "b"
+
+
```

Comment:
Let's also add a test for use cases when a `MultiLayoutPage` subclass also uses multiple `MultiLayoutPage`s underneath. I think this could be a use case when users import POs from different packages and repacking them.

```python
class MyMultiLayoutPage(MultiLayoutPage[SomeItem]):
    response: HttpResponse
    layout_page_us: LayoutPageUS
    layout_page_uk: LayoutPageUK

    async def get_layout(self) -> ItemPage[SomeItem]:
        if self.response.css(".origin::text") == "us":
            return self.layout_page_us.get_layout()
        return self.layout_page_uk.get_layout()
```

Might also be worth creating a doc about this as well.